design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+The ClusterLink components will be created in `clusterlink-system-ns`, and the control-plane will have privileged permissions within the namespace.
+Additionally, the operator will create and update components in response to the CRD instance create or update actions. Once created or updated, the operator will not actively monitor the state of each component, assuming that any changes will be made only by privileged users.
+Furthermore, the operator will delete all components and the namespace of ClusterLink when the CRD instance is deleted.
+The ClusterLink deployment is limited to one instance per cluster. In the case of deploying more than one CRD instance, only the first one will be taken into account, and the others will be ignored.
+
+The deployment file for the operator and example of CRD instance will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.
+For instance, the following commands can assist the fabric administrator in creating the site certificates:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+For the site administrator, automation of deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the CRD instance can be achieved with:
+
+    clusterlink install --site <site_name>
+
+Note: The ClusterLink CLI uses the kubeconfig, and it should be set to the specific cluster that needs to be configured
+
+### ClusterLink CRD
+
+The ClusterLink CRD includes the following fields:
+
+* **API version:** clusterlink.net/v1alpha1
+* **Kind:** clusterlink
+
+* **Spec:**
+  
+    | Field name | Subfield name| Description | Default value |
+    | ---- | ----- | ------ | ----|
+    |dataplane | | ClusterLink dataplane component attributes||
+    |  |type| Types of dataplane, supported values "go" or "envoy"|"envoy"|
+    | |replicates| Number of dataplane replicas|1|
+    |ingress|| ClusterLink ingress component attributes ||
+    ||type| Type of service to expose ClusterLink deployment, supported values: "load-balancer", "API-gateway",    "node-port". |load-balancer|
+    |logLevel| |Log level severity for all the components (controlplane and dataplane)| "info"|
+    |containerRegistry| |The container registry to pull the project images when the images is not present locally | ghcr.io/clusterlink-net|
+    |imageTag| |The project images version | latest|
+
+* **Status:**
+
+    | Field name | Description | Default value |
+    | ----------- | ----------- | ------------- |
+    | controlplaneStatus      | Status of the controlplane components controlled by the operator. Supported values: <br> None - if all components have not been created. <br> Ready - if all components are in a ready and running status. <br> Error - If one of the components is in an error status. <br> Update - if one of the components is in an update status. | None |
```

Comment:
- No need for XYZStatus field name as it is implied by being under the `Status` object
- Consider using k8s Conditions